src/implementationGuides.ts

```diff
@@ -0,0 +1,18 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+export interface ImplementationGuides {
+    /**
+     * Compiles the contents of an Implementation Guide into an internal representation used to implement the features specified by it.
+     *
+     * The FHIR resource types commonly used on Implementation Guides are: StructureDefinition, SearchParameter, CodeSystem, ValueSet, ImplementationGuide, ConceptMap, OperationDefinition, CapabilityStatement.
+     *
+     * Different implementations of this interface may choose to process only a subset of the above resources.
+     *
+     * @param input - an array of FHIR resource objects to be compiled.
+     * @return compiled output - The output is not meant to be used by any other entity other than the module that implements this interface.
+     */
+    compile(input: any[]): Promise<any>;
```

Comment:
Is the compile function supposed to take all FHIR conformance resources at once or can we make multiple calls with a smaller array of resources each, let's say once for each IG ?

Reply:
The idea is to take all FHIR conformance resources from all IGs in scope at once. 

Calling it once for each IG won't cause an error, but it's not practical, since the caller does not know how to combine the results into a single object that can be used by FWoA (i.e. the search compiler may know how to combine SearchParameters with the same name from different IGs, but the caller may not know how to combine the compiled outputs)